projects/plugins/jetpack/extensions/blocks/donations/amount.js

```diff
@@ -78,6 +78,11 @@ const Amount = ( {
 		setEditedValue( formatCurrency( value, currency, { symbol: '' } ) );
 	}, [ currency, isFocused, isInvalid, setAmount, value ] );
 
+	useEffect( () => {
+		setAmount( formatCurrency( value, currency, { symbol: '' } ) );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ value ] );
```

Comment:
I think removing the dependencies would run this when the component is mounted, therefore achieving the expected effect, and you could remove the `// eslint-disable-next-line react-hooks/exhaustive-deps`.

```suggestion
	} );
```

Reply:
This doesn't actually fix the issue! (I know, right!) The problem here is the Amount component doesn't have the normal react lifecycle.  It's actually the same component on each tab, so the initial effects only apply when the block is first loaded.  When you change the tab it's not recreating the Amount it's just changing the amounts shown.  I tried the exhaustive list too (to remove the line), but that creates an infinite loop.  We really only want to run this when the value has changed, I can apply this change:

```diff
diff --git a/projects/plugins/jetpack/extensions/blocks/donations/amount.js b/projects/plugins/jetpack/extensions/blocks/donations/amount.js
index 7479565180..1dd8dd8397 100644
--- a/projects/plugins/jetpack/extensions/blocks/donations/amount.js
+++ b/projects/plugins/jetpack/extensions/blocks/donations/amount.js
@@ -4,6 +4,14 @@ import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
 import classnames from 'classnames';
 import { minimumTransactionAmountForCurrency, parseAmount } from '../../shared/currencies';
 
+function usePreviousValue( value ) {
+       const ref = useRef();
+       useEffect( () => {
+               ref.current = value;
+       } );
+       return ref.current;
+}
+
 const Amount = ( {
        className = null,
        currency = null,
@@ -19,6 +27,7 @@ const Amount = ( {
        const [ isFocused, setIsFocused ] = useState( false );
        const [ isInvalid, setIsInvalid ] = useState( false );
        const richTextRef = useRef( null );
+       const previousValue = usePreviousValue( value );
 
        const setAmount = useCallback(
                amount => {
@@ -79,9 +88,10 @@ const Amount = ( {
        }, [ currency, isFocused, isInvalid, setAmount, value ] );
 
        useEffect( () => {
-               setAmount( formatCurrency( value, currency, { symbol: '' } ) );
-               // eslint-disable-next-line react-hooks/exhaustive-deps
-       }, [ value ] );
+               if ( previousValue !== value ) {
+                       setAmount( formatCurrency( value, currency, { symbol: '' } ) );
+               }
+       }, [ currency, previousValue, setAmount, value ] );
 
        return (
                <div
```

What do you think?

Reply:
> This doesn't actually fix the issue! (I know, right!) 

I was sure I tried and it was working...  Damn... 

> What do you think?

I am not that familiar with `useRef()`... So I trust you if you think it works...
Otherwise leave the actual one with the es-lint comment. I think it reads better than this version...

Also, there are projects in motion to refactor this whole block and merge it with the Team51 one: p81Rsd-1n8-p2
and pewaj1-kC-p2#comment-252, so it does not matter too much if we keep the disable es-lint comment :) ...

I also think this issue is such a small one it is not worth spending more time on it....